packages/resource-deployment/scripts/batch-account-create.sh

```diff
@@ -48,53 +90,16 @@ if ! az account show 1>/dev/null; then
     az login
 fi
 
+. "${0%/*}/set-resource-names.sh"
+
 # Configure Azure subscription account to support Batch account in user subscription mode
 . "${0%/*}/account-set-batch-app.sh"
 
-# Deploy Azure Batch account using resource manager template
-echo "Deploying Azure Batch account in resource group $resourceGroupName with template $batchTemplateFile"
-resources=$(
-    az group deployment create \
-        --resource-group "$resourceGroupName" \
-        --template-file "$batchTemplateFile" \
-        --query "properties.outputResources[].id" \
-        --parameters enableSoftDeleteOnKeyVault="$enableSoftDeleteOnKeyVault" \
-        -o tsv
-)
-
-# Get key vault and batch account resources
-. "${0%/*}/get-resource-name-from-resource-paths.sh" -p "Microsoft.KeyVault/vaults" -r "$resources"
-keyVault="$resourceName"
-
-. "${0%/*}/get-resource-name-from-resource-paths.sh" -p "Microsoft.Batch/batchAccounts" -r "$resources"
-batchAccountName="$resourceName"
-
-if [[ -z $batchAccountName ]] || [[ -z $keyVault ]]; then
-    echo \
-        "Unable to get required resource information from Batch account deployment:
-    batchAccountName - $batchAccountName
-    keyVault - $keyVault"
+deployBatch
 
-    exit 1
-fi
-
-if [[ -z $enableSoftDeleteOnKeyVault ]]; then
-    enableSoftDeleteOnKeyVault=true
-fi
-
-# Login into Azure Batch account
-echo "Logging into '$batchAccountName' Azure Batch account"
-az batch account login --name "$batchAccountName" --resource-group "$resourceGroupName"
+. "${0%/*}/push-secrets-to-key-vault.sh"
```

Comment:
not logically belongs to this script, can we have it in top level script?

Reply:
keyvault is created by batch. Also, for batch nodes to be up & running, we need keyvault to be setup.
having this here so that, nodes don't startup with failed tasks & run till timeout.

Reply:
you can still have it in setup-all- script

Reply:
moving it to setup-all-pools-for-batch.sh will require us to pass the secrets to that script, even though it has nothing to do with keyvault. (setup-all-pools doesnt create keyvault.) batch account template is the one which creates the keyvault.

Reply:
I see batch-account-create.sh is only called from install-parallel.sh, the latter had the push-secrets-to-key-vault.sh but beeing moved under former along w/ secrets, so keeping original version possible and no changes required for this pr right?

Reply:
I did this change when i was working on enabling monitor. I had to wait for pools to be stable (finish startup task) so that it wont throw lock issues when installing monitoring agent. Waiting for pool nodes will be longer if key vault secrets are not copied, since we have a retry mechanism in startup script. Lets discuss offline, if you still feel this should be outside. I think, if copying secrets should be outside, then so does the batch-pool-setup call.